### PR TITLE
Organize doc menu into groups

### PR DESCRIPTION
**Pull request by levithomason**
image

### CODE REVIEW

File: docs/app/Components/Sidebar/Sidebar.js [hunk 1]
```diff
@@ -1,28 +1,65 @@
...
   handleSearchChange = e => this.setState({query: e.target.value});
 
+  getComponentsByQuery() {
+    return _.filter(stardust, component => {
+      const name = component._meta.name;
+      const isParent = META.isParent(component);
+      const isQueryMatch = new RegExp(this.state.query, 'i').test(name);
+      return isParent && isQueryMatch;
+    });
+  }
```
Comment (levithomason) [inline docs/app/Components/Sidebar/Sidebar.js:18]: Pulled menu item filtering into separate methods.

File: docs/app/Components/Sidebar/Sidebar.js [hunk 2]
```diff
@@ -1,28 +1,63 @@
...
+
+  getComponentsByType = type => {
```
Comment (levithomason) [inline docs/app/Components/Sidebar/Sidebar.js:20]: This method builds a submenu for each section, like:

**Elements**
- Button

Only parent elements are included in the menu.  For instance, Modal, but not its child components ModalHeader, ModalContent, etc.

File: docs/app/Components/Sidebar/Sidebar.js [hunk 3]
```diff
@@ -1,28 +1,63 @@
...
             placeholder='Search'
             iconClass='search link icon'
             onChange={this.handleSearchChange}
           />
         </MenuItem>
-        {menuItems}
+        {elements}
+        {collections}
+        {views}
+        {modules}
```
Comment (levithomason) [inline docs/app/Components/Sidebar/Sidebar.js:60]: Order all the submenus.

File: src/collections/Menu/Menu.js [hunk 4]
```diff
@@ -34,7 +34,7 @@ export default class Menu extends Component {
       const activeItemName = !hasActiveItem && i === 0
         ? child.props.name
         : this.state.activeItem || this.props.activeItem;
-      return React.cloneElement(child, {
+      return child && React.cloneElement(child, {
```
Comment (levithomason) [inline src/collections/Menu/Menu.js:37]: The menu component cannot clone the child if it doesn't exist.  When filtering the doc site menu by search we we're getting errors.  Now, we skip cloning and updating menu children if the child is falsy.

File: src/collections/Menu/MenuItem.js [hunk 5]
```diff
@@ -36,7 +36,6 @@ export default class MenuItem extends Component {
     const classes = classNames(
       'sd-menu-item',
       this.props.className,
-      'blue',
```
Comment (levithomason) [inline src/collections/Menu/MenuItem.js:39]: Removed default color class on menu item, this should be up to the consumer.

File: src/elements/Button/Buttons.js [hunk 6]
```diff
@@ -12,6 +12,7 @@ export default class Buttons extends Component {
     library: META.library.semanticUI,
     name: 'Buttons',
     type: META.type.element,
+    parent: 'Button'
```
Comment (levithomason) [inline src/elements/Button/Buttons.js:15]: Buttons is not a top level component, but a child to the Button component.   This prevents it from showing in the menu.

File: src/utils/Meta.js [hunk 7]
```diff
@@ -28,9 +30,14 @@ const META = {
...
   isModule: ({_meta}) => _meta.type === META.type.module,
+  isType: ({_meta}, type) => _meta.type === type,
 
   // parent
-  isChild: ({_meta}) => !!_meta.parent,
+  isParent: (component) => (
+    // has no 'parent' or 'parent' is self
+    !_.has(component, '_meta.parent') || _.get(component, '_meta.parent') === component._meta.name
+  ),
+  isChild: component => !META.isParent(component),
```
Comment (levithomason) [inline src/utils/Meta.js:40]: Extended META to include helper method for identifying a parent component and a component with a type string.

**Comment by eanplatter** [comment]
:fallen_leaf:

**Pull request merged**